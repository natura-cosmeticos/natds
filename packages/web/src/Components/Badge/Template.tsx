import * as React from 'react'

import { Story } from '@storybook/react'
import makeStyles from '@material-ui/core/styles/makeStyles'
import createStyles from '@material-ui/core/styles/createStyles'
import clsx from 'clsx'
import { IBadgeProps } from './Badge.props'
import { IThemeWeb } from '../../Themes'
import Badge from './Badge'
import * as IconStories from '../Icon/Icon.stories'
import { Icon } from '../Icon'

const useStyles = makeStyles((theme: IThemeWeb) => createStyles({
  root: {
    '& > *': {
      marginRight: theme.sizes.small
    }
  },
  shape: {
    backgroundColor: '#FF00FF1A',
    height: theme.sizes.semiX as number,
    width: theme.sizes.semiX as number
  },
  shapeCircle: {
    borderRadius: theme.shape ? theme.shape.badgeBorderRadius : ''
  }
}), { name: 'BadgeDemo' })

export const BadgeShapeCircleContent : Story<IBadgeProps> = (props: IBadgeProps) => {
  const { shape, shapeCircle } = useStyles()

  return <div className={clsx(shape, shapeCircle)} {...props} />
}

export const BadgeShapeContent : Story<IBadgeProps> = (props: IBadgeProps) => {
  const { shape } = useStyles()

  return <div className={shape} {...props} />
}

export const Template : Story<IBadgeProps> = (args: IBadgeProps) => (
  <Badge {...args}>
    {args.children && args.children}
    {!args.children && (args.overlap === 'circle' ? <BadgeShapeCircleContent /> : <BadgeShapeContent />)}
  </Badge>
)

export const TemplateWithShapes : Story<IBadgeProps> = (args: IBadgeProps) => {
  const { root } = useStyles()

  return (
    <div className={root}>
      <Badge {...args}>
        <BadgeShapeContent />
      </Badge>
      <Badge {...args}>
        <BadgeShapeCircleContent />
      </Badge>
      <Badge {...args}>
        {args.children || <Icon name="outlined-communication-email" />}
      </Badge>
    </div>
  )
}
export default Template
