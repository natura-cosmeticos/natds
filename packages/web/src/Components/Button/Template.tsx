import { Story } from '@storybook/react'
import * as React from 'react'
import { IButtonProps } from './Button.props'
import { Button } from './Button'

export const Template : Story<IButtonProps> = (args) => <Button {...args} />

export const TemplateWithColors : Story<IButtonProps> = (args) => (
  <>
    <Button {...args} color="primary">Primary color</Button>
&nbsp;
    <Button {...args} color="secondary">Secondary color</Button>
&nbsp;
    <Button {...args} color="default">Default color</Button>
&nbsp;
  </>
)

export const TemplateWithSizes : Story<IButtonProps> = (args) => (
  <>
    <Button {...args} size="small">Small size</Button>
&nbsp;
    <Button {...args} size="medium">Medium size</Button>
&nbsp;
    <Button {...args} size="large">Large size</Button>
&nbsp;
  </>
)

export default Template
