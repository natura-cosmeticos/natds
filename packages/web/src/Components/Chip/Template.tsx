import { Story } from '@storybook/react'
import * as React from 'react'
import { IChipProps } from './Chip.props'
import { Chip } from './Chip'
import { variants } from './__fixtures__/variants'

export const Template : Story<IChipProps> = (args: IChipProps) => <Chip {...args} />

export const TemplateWithVariants : Story<IChipProps> = (args: IChipProps) => (
  <div>
    <Chip {...args} variant={variants.default} />
    {' '}
    <Chip {...args} variant={variants.outlined} />
  </div>
)

export default Template
