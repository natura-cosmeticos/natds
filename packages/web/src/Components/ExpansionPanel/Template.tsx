import { Story } from '@storybook/react'
import * as React from 'react'
import { IExpansionPanelProps } from './ExpansionPanel.props'
import { ExpansionPanel } from './ExpansionPanel'
import { ExpansionPanelSummary } from '../ExpansionPanelSummary'
import {
  Playground as SummaryPlayground,
  WithTypography, WithTypographyInsideDiv
} from '../ExpansionPanelSummary/ExpansionPanelSummary.stories'
import { ExpansionPanelDetails } from '../ExpansionPanelDetails'
import {
  Playground as DetailsPlayground,
  WithSubtitle1, WithSubtitle1AndTextPrimaryColor,
  WithSubtitle1AndTextSecondaryColor
} from '../ExpansionPanelDetails/ExpansionPanelDetails.stories'
import { ExpansionPanelActions } from '../ExpansionPanelActions'
import { Playground as ActionsPlayground } from '../ExpansionPanelActions/ExpansionPanelActions.stories'
import { Divider } from '../Divider'
import { Playground as DividerPlayground } from '../Divider/Divider.stories'

export const Template : Story<IExpansionPanelProps> = (args: IExpansionPanelProps) => (
  <div>
    <ExpansionPanel {...args}>
      <ExpansionPanelSummary {...SummaryPlayground.args} />
      <ExpansionPanelDetails {...WithSubtitle1.args} />
    </ExpansionPanel>
    <ExpansionPanel {...args}>
      <ExpansionPanelSummary {...SummaryPlayground.args} />
      <ExpansionPanelDetails {...DetailsPlayground.args} />
      <ExpansionPanelActions {...ActionsPlayground.args} />
    </ExpansionPanel>
    <ExpansionPanel {...args}>
      <ExpansionPanelSummary {...WithTypography.args} />
      <ExpansionPanelDetails {...WithSubtitle1AndTextSecondaryColor.args} />
    </ExpansionPanel>
    <ExpansionPanel {...args}>
      <ExpansionPanelSummary {...WithTypography.args} />
      <ExpansionPanelDetails {...WithSubtitle1AndTextPrimaryColor.args} />
    </ExpansionPanel>
    <ExpansionPanel {...args}>
      <ExpansionPanelSummary {...WithTypography.args} />
      <ExpansionPanelDetails {...WithSubtitle1AndTextPrimaryColor.args} />
      <ExpansionPanelActions {...ActionsPlayground.args} />
    </ExpansionPanel>
    <ExpansionPanel {...args}>
      <ExpansionPanelSummary {...WithTypographyInsideDiv.args} />
      <ExpansionPanelDetails {...WithSubtitle1AndTextPrimaryColor.args} />
      <Divider {...DividerPlayground.args} />
    </ExpansionPanel>
  </div>
)

export default Template
