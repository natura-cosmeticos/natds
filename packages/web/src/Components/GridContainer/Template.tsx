import makeStyles from "@material-ui/core/styles/makeStyles";
import createStyles from "@material-ui/core/styles/createStyles";
import * as React from "react";
import { Story } from "@storybook/react";
import { IGridContainerProps } from "./GridContainer.props";
import { ITypographyProps } from "../Typography";
import { GridItem } from "../GridItem";
import { Card } from "../Card";

const GridContainer : React.ForwardRefExoticComponent<IGridContainerProps> = React.lazy(() => import("./GridContainer"));
const Typography : React.ForwardRefExoticComponent<ITypographyProps> = React.lazy(() => import("../Typography"));

GridContainer.displayName = "GridContainer";
Card.displayName = "Card";
Typography.displayName = "Typography";

export const Template : Story<IGridContainerProps> = (args) => {
  const useStyles = makeStyles(createStyles({
    container: {
      backgroundColor: "#FF00FF1A",
    },
    item: {
      backgroundColor: "#FF00FF1A",
      boxSizing: "border-box",
      textAlign: "center",
    },
  }), { name: "GridDemo" });

  const classes = useStyles();

  return <GridContainer className={classes.container} {...args}>
    <GridItem className={classes.item} lg={12} md={12} sm={12} xs={12}>
      <Card elevation={2}>{args.children}</Card>
    </GridItem>
    <GridItem className={classes.item} md={6} sm={6} xs={6}>
      <Card elevation={2}>{args.children}</Card>
    </GridItem>
    <GridItem className={classes.item} md={6} sm={6} xs={6}>
      <Card elevation={2}>{args.children}</Card>
    </GridItem>
  </GridContainer>;

};

export default Template;
