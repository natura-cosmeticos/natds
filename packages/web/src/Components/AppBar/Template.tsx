import { Story } from '@storybook/react'
import * as React from 'react'
import { IAppBarProps } from './AppBar.props'
import { IToolbarProps, Toolbar } from '../Toolbar'
import { AppBar } from './AppBar'
import { Typography } from '../Typography'
import { TabContainer } from '../TabContainer'
import { ForAppBar } from '../TabContainer/TabContainer.stories'
import { TabItem } from '../TabItem'
import {
  WithBrandsProductIconAndLabel,
  WithLoveIconAndLabel,
  WithMakeupIconAndLabel,
  WithOutletIconAndLabel
} from '../TabItem/TabItem.stories'
import { Spacing } from '../Spacing'
import { SmallRightMargin } from '../Spacing/Spacing.margin.stories'
import { IconButton } from '../IconButton'
import { Icon } from '../Icon'
import { Button } from '../Button'
import { InheritColor } from '../Button/Button.stories'
import { positions } from './__fixtures__/positions'
import {
  FilterButtonExample,
  MenuButtonExample,
  MyProfileButtonExample,
  SearchButtonExample
} from '../IconButton/IconButton.stories'
import { AppBarHighlight } from '../AppBarHighlight'
import { ForAppBarTitle } from '../Typography/Typography.stories'
import { SmallIcon } from '../Icon/Icon.stories'

export const Template : Story<IAppBarProps> = (args: IAppBarProps) => (
  <AppBar {...args}>
    <Toolbar>
      <AppBarHighlight>
        <Typography {...ForAppBarTitle.args}>{args.children}</Typography>
      </AppBarHighlight>
    </Toolbar>
  </AppBar>
)

export const TemplateWithTabs : Story<IAppBarProps> = (args: IAppBarProps) => {
  const initialState = 0
  const [value, setValue] = React.useState<number>(initialState)

  // eslint-disable-next-line @typescript-eslint/ban-types
  const handleChange = (event: React.ChangeEvent<{}>, newValue: number) => {
    setValue(newValue)
  }

  return (
    <AppBar {...args}>
      {/* @ts-ignore */}
      <TabContainer {...ForAppBar.args} onChange={handleChange} value={value}>
        <TabItem {...WithLoveIconAndLabel.args} />
        <TabItem {...WithOutletIconAndLabel.args} />
        <TabItem {...WithBrandsProductIconAndLabel.args} />
        <TabItem {...WithMakeupIconAndLabel.args} />
      </TabContainer>
    </AppBar>
  )
}

export const ToolbarTemplate : Story<IToolbarProps> = (args: IToolbarProps) => (
  <AppBar position={positions.static}>
    <Toolbar {...args}>
      <Spacing {...SmallRightMargin.args}>
        <IconButton>
          <Icon {...SmallIcon.args} name="filled-brand-naturarosacea" />
        </IconButton>
      </Spacing>
      {args.children}
      <Button {...InheritColor.args}>Login</Button>
    </Toolbar>
  </AppBar>
)

export const TemplateWithIconButtons : Story<IAppBarProps> = (args: IAppBarProps) => (
  <AppBar {...args}>
    <Toolbar>
      <IconButton {...MenuButtonExample.args} />
      <AppBarHighlight>
        <Typography {...ForAppBarTitle.args}>{args.children}</Typography>
      </AppBarHighlight>
      <IconButton {...SearchButtonExample.args} />
      <IconButton {...FilterButtonExample.args} />
      <IconButton {...MyProfileButtonExample.args} />
    </Toolbar>
  </AppBar>
)
