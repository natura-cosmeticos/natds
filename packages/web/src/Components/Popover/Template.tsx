import { Story } from '@storybook/react'
// eslint-disable-next-line no-use-before-define
import * as React from 'react'
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
import { IPopoverProps } from './Popover.props'
import { Spacing } from '../Spacing'
import { VeryHugeMargin } from '../Spacing/Spacing.margin.stories'
import { Button } from '../Button'
import { ContainedVariant } from '../Button/Button.stories'
import { Popover } from './Popover'

export const Template : Story<IPopoverProps> = (args: IPopoverProps) => {
  const [anchor, setAnchor] = React.useState<IPopoverProps['anchorEl']>()
  const [open, setOpen] = React.useState<IPopoverProps['open']>(args.open)

  const handleButtonClick = () => {
    setOpen((previousValue) => !previousValue)
  }

  return (
    <Spacing display="flex" {...VeryHugeMargin.args}>
      <Button
        {...ContainedVariant.args}
        aria-describedby={args.id}
        onClick={handleButtonClick}
        ref={setAnchor}
      >
        Toggle Popover
      </Button>
      <Popover
        {...args}
        anchorEl={anchor}
        id={args.id}
        open={open}
        transition={args.transition}
      >
        {args.children}
      </Popover>
    </Spacing>
  )
}

export default Template
