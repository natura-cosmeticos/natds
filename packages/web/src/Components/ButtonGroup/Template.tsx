import { Story } from '@storybook/react'
import * as React from 'react'
import { IButtonGroupProps } from './ButtonGroup.props'
import { ButtonGroup } from './ButtonGroup'
import { Button } from '../Button'
import { Icon } from '../Icon'
import { IconButton } from '../IconButton'
import { colors } from './__fixtures__/colors'
import { Typography } from '../Typography'
import { Caption } from '../Typography/Typography.stories'
import { GridContainer } from '../GridContainer'
import { GridItem } from '../GridItem'
import { Spacing3 } from '../GridContainer/GridContainer.stories'
import { variants } from './__fixtures__/variants'
import { TinyIcon } from '../Icon/Icon.stories'

export const Template : Story<IButtonGroupProps> = (args: IButtonGroupProps) => (
  <ButtonGroup {...args}>
    <Button>Add</Button>
    <Button>Edit</Button>
    <Button>Delete</Button>
  </ButtonGroup>
)

export const TemplateWithIcons : Story<IButtonGroupProps> = (args: IButtonGroupProps) => (
  <ButtonGroup {...args}>
    <Button aria-label="Add">
      <Icon {...TinyIcon.args} name="filled-action-add" />
    </Button>
    <Button aria-label="Edit">
      <Icon {...TinyIcon.args} name="outlined-action-edit" />
    </Button>
    <Button aria-label="Delete">
      <Icon {...TinyIcon.args} name="outlined-action-delete" />
    </Button>
  </ButtonGroup>
)

export const TemplateWithIconButtons : Story<IButtonGroupProps> = (args: IButtonGroupProps) => (
  <ButtonGroup {...args}>
    <IconButton aria-label="Add">
      <Icon {...TinyIcon.args} name="filled-action-add" />
    </IconButton>
    ,
    <IconButton aria-label="Edit">
      <Icon {...TinyIcon.args} name="outlined-action-edit" />
    </IconButton>
    ,
    <IconButton aria-label="Delete">
      <Icon {...TinyIcon.args} name="outlined-action-delete" />
    </IconButton>
  </ButtonGroup>
)

export const TemplateWithColors : Story<IButtonGroupProps> = (args: IButtonGroupProps) => (
  <GridContainer {...Spacing3.args}>
    {Object.keys(colors).map((color: IButtonGroupProps['color'], index) => (
      <GridItem key={index}>
        <Typography component="h4" paragraph {...Caption.args}>{color}</Typography>
        <ButtonGroup {...args} color={color}>
          <Button>Add</Button>
          <Button>Edit</Button>
          <Button>Delete</Button>
        </ButtonGroup>
      </GridItem>
    ))}
  </GridContainer>
)

export const TemplateWithVariants : Story<IButtonGroupProps> = (args: IButtonGroupProps) => (
  <GridContainer {...Spacing3.args}>
    {Object.keys(variants).map((variant: IButtonGroupProps['variant'], index) => (
      <GridItem key={index}>
        <Typography component="h4" paragraph {...Caption.args}>{variant}</Typography>
        <ButtonGroup {...args} variant={variant}>
          <Button>Add</Button>
          <Button>Edit</Button>
          <Button>Delete</Button>
        </ButtonGroup>
      </GridItem>
    ))}
  </GridContainer>
)

export default Template
