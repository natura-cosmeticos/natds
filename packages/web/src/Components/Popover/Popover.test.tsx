import * as TestRenderer from 'react-test-renderer'
// eslint-disable-next-line no-use-before-define
import * as React from 'react'
import { Template } from './Template'
import { createNodeMock } from '../../__mocks__/createNodeMock'

describe('Popover component', () => {
  describe('when closed', () => {
    it('should match snapshot', () => {
      const testRenderer : TestRenderer.ReactTestRenderer = TestRenderer.create(
        <Template open={false}>Closed popover</Template>
      )

      expect(testRenderer).toMatchSnapshot()
    })
  })
})
