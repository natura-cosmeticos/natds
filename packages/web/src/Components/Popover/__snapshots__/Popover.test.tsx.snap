// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover component when closed should match snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 NatDSSpacing-root-1 NatDSSpacing-root-2"
>
  <button
    className="MuiButtonBase-root MuiButton-root NatDSButton-root-4 MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "textTransform": undefined,
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label NatDSButton-label-5"
    >
      Toggle Popover
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;
